### PR TITLE
[13.0][FIX] project_stock: Change cancel process (set as quantity_done = 0)

### PR DESCRIPTION
**Pull request by victoralmau**
Only cancel the stock moves not done

Please @pedrobaeza and @sergio-teruel can you review it?

@Tecnativa TT39138

### CODE REVIEW

File: project_stock/models/project_task.py [hunk 1]
```diff
@@ -90,12 +90,14 @@ def _compute_scrap_move_count(self):
         for item in self:
             item.scrap_count = count_data.get(item.id, 0)
 
+    @api.depends("move_ids.state")
```
Comment (pedrobaeza) [inline project_stock/models/project_task.py]: For additions and removals

```suggestion
    @api.depends("move_ids", "move_ids.state")
```

File: project_stock/models/project_task.py [hunk 2]
```diff
@@ -196,7 +198,7 @@ def action_cancel(self):
         """Cancel the stock moves and remove the analytic lines created from
         stock moves when cancelling the task.
         """
-        self.move_ids.write({"state": "cancel"})
+        self.move_ids.write({"quantity_done": 0})
```
Comment (pedrobaeza) [inline project_stock/models/project_task.py]: Don't you cancel the move afterwards?

Comment (victoralmau) [reply project_stock/models/project_task.py]: Cancelling a stock move is not possible with UX and not allowed by odoo https://github.com/odoo/odoo/blob/13.0/addons/stock/models/stock_move.py#L1324

Defining as quantity done 0 is the most optimal way to do this "_cancellation_" process. Also, this change does re-generate correctly `stock.quant` records.

**Review by pedrobaeza — APPROVED** [review]

**Comment by victoralmau** [comment]
Ping @sergio-teruel

**Review by sergio-teruel — APPROVED** [review]

**Pull request merged**